Update Travis-CI configuration

Arcanemagus opened the pull request:
Includes:

* Building on both stable and beta channels of Atom.
* Proper execution of the build script

Arcanemagus committed:
Update Travis-CI configuration

Includes:

* Building on both stable and beta channels of Atom.
* Proper execution of the build script

norio-nomura commented on .travis.yml:
```diff
@@ -1,9 +1,28 @@
+    - ATOM_CHANNEL=beta
+
+install:
+  - brew update && brew install swiftlint
```
swiftlint is pre-installed on Travis. So we should use:
```yaml
before_install:
  - brew update
  - brew outdated swiftlint || brew upgrade swiftlint
```
https://docs.travis-ci.com/user/osx-ci-environment/#A-note-on-upgrading-packages

norio-nomura commented:
Default image of `language: objective-c` is macOS 10.11 with Xcode 7.3.1.
We should add `osx_image: xcode8.2` for avoiding error on #44.

Arcanemagus committed:
Properly update swiftlint

`swiftlint` is already installed on the Travis-CI macOS image now,
change the script to update it instead of attempting to install it
(which would fail due to how `brew` works).

Arcanemagus committed:
Update the macOS image used

We need to use the latest version of XCode to run `swqiftlint`, force
the XCode 8.2 image.

Arcanemagus commented:
Pushed up your suggested changes, let's see how this goes, thanks!

norio-nomura commented:
We have also changed the runtime requirement of SwiftLint in Homebrew distribution.
https://github.com/Homebrew/homebrew-core/pull/9989
Now, SwiftLint can be installed by using Homebrew binary package with Xcode 7.0 or later.

Arcanemagus commented:
Good to know, so at some point [this line](https://github.com/AtomLinter/linter-swiftlint/blob/54c75eb43d523f5700351a7257110dce210dc2cf/.travis.yml#L3) could be removed.
